aspnetcore/fundamentals/hosted-services/sample/Services/TimedHostedService.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace BackgroundTasksSample.Services
+{
+    #region snippet1
+    internal class TimedHostedService : IHostedService, IDisposable
+    {
+        private Timer _timer;
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is starting.");
+
+            _timer = new Timer(DoWork, null, TimeSpan.Zero, TimeSpan.FromSeconds(5));
+
+            return Task.CompletedTask;
+        }
+
+        private void DoWork(object state)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is working.");
+        }
+
+        public Task StopAsync(CancellationToken cancellationToken)
+        {
+            Console.WriteLine($"{DateTime.UtcNow} - Timed Background Service is stopping.");
+
+            _timer.Change(Timeout.Infinite, 0);
+
+            return Task.CompletedTask;
+        }
+
+        public void Dispose()
+        {
+            _timer.Dispose();
```

Comment:
You need to handle the timer being null.

Reply:
"handle" ... ?? 😕 Chris said to do it this way. Zero (kill) the timer on `StopAsync` then dispose of it on `Dispose`.

Reply:
What if the host never starts?

Reply:
Ur saying this will fail if the host never starts? ... will it throw an ORE or something?

Reply:
`_timer?.Dispose();` It is possible StartAsync is never called. E.g. if another service throws from it's StartAsync.

Reply:
Correct. If you assign something in start then you have to take care of StartAsync never being called on StopAsync and Dispose